src/models/shelf/spec.ts

```diff
@@ -27,28 +26,31 @@ export interface ShelfUnitSpec {
   anyEncodings: ShelfAnyEncodingDef[];
 
   config: Config;
+
+  filters: Array<RangeFilter | OneOfFilter>;
```

Comment:
Can you move `filters` and `transform` to be above `mark` and `encoding` for all of these?